plugins/fluent-bit/builtins/plugin.cpp

```diff
@@ -0,0 +1,140 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "fluent-bit/fluent-bit_operator.hpp"
+
+namespace tenzir::plugins::fluentbit {
+
+namespace {
+
+class from_fluent_bit_plugin final
+  : public operator_plugin2<fluent_bit_source_operator> {
+public:
+  auto name() const -> std::string override {
+    return "from_fluent_bit";
+  }
+  auto initialize(const record& unused_plugin_config,
+                  const record& global_config) -> caf::error override {
+    if (not unused_plugin_config.empty()) {
+      return diagnostic::error("`{}.yaml` is unused; Use `fluent-bit.yaml` "
+                               "instead",
+                               this->name())
+        .to_error();
+    }
+    auto c = try_get_only<tenzir::record>(global_config, "plugins.fluent-bit");
+    if (not c) {
+      return c.error();
+    }
+    if (*c) {
+      config_ = **c;
+    }
+    return caf::none;
+  }
+
+  auto
+  make(invocation inv, session ctx) const -> failure_or<operator_ptr> override {
+    auto parser = argument_parser2::operator_(name());
+    located<std::string> plugin;
+    std::optional<tenzir::record> plugin_options;
+    std::optional<tenzir::record> fluentbit_options;
+    parser.positional("plugin", plugin)
+      .named("options", plugin_options)
+      .named("fluent_bit_options", fluentbit_options);
+    auto opt_parser = multi_series_builder_argument_parser{};
+    opt_parser.add_all_to_parser(parser);
+    auto result = parser.parse(inv, ctx);
+    TRY(result);
+    auto args = operator_args{
+      .plugin = std::move(plugin),
+      .service_properties = to_property_map(fluentbit_options),
+      .args = to_property_map(plugin_options),
+    };
+    TRY(auto builder_options, opt_parser.get_options(ctx.dh()));
+    builder_options.settings.default_schema_name
+      = fmt::format("fluent_bit.{}", args.plugin.inner);
+    return std::make_unique<fluent_bit_source_operator>(
+      std::move(args), std::move(builder_options), config_);
+  }
+
+  virtual auto load_properties() const -> load_properties_t override {
+    return {
+      .schemes = {"fluentbit"},
+      .accepts_pipeline = false,
+      .strip_scheme = true,
+      .events = true,
+      .transform_uri = {},
+    };
+  }
+
+private:
+  record config_;
+};
+
+class to_fluent_bit_plugin final
+  : public operator_plugin2<fluent_bit_sink_operator> {
+public:
+  auto name() const -> std::string override {
+    return "to_fluent_bit";
+  }
```

Comment:
Btw, you don't need to override the name if you inherit from `operator_plugin2`.